src/ui/theme/utils.ts

```diff
@@ -0,0 +1,224 @@
+import { type BrandingAppearance } from "src/networking/responses/branding-response";
+import {
+  type Colors,
+  DEFAULT_FORM_COLORS,
+  DEFAULT_INFO_COLORS,
+  FormColorsToBrandingAppearanceMapping,
+  InfoColorsToBrandingAppearanceMapping,
+} from "./colors";
+import {
+  DefaultShape,
+  PillsShape,
+  RectangularShape,
+  RoundedShape,
+  type Shape,
+} from "./shapes";
+
+type RGB = {
+  r: number;
+  g: number;
+  b: number;
+};
+
+const hexToRGB = (color: string): RGB | null => {
+  if (color.length == 7)
+    return {
+      r: parseInt(color.slice(1, 3), 16),
+      g: parseInt(color.slice(3, 5), 16),
+      b: parseInt(color.slice(5, 7), 16),
+    };
+  if (color.length == 4)
```

Comment:
Hmm should we always require a `#RRGGBB` format? Not sure if we need to support this as well...

Reply:
It's quite typical to shorthand a color with `#fff` I'd keep it if possible.